lib/Site.js

```diff
@@ -250,7 +278,9 @@ Site.prototype.generate = function (baseUrl) {
       .then(() => this.collectBaseUrl())
       .then(() => this.collectUserDefinedVariablesMap())
       .then(() => this.buildAssets())
+      .then(() => this.collectAddressablePaths())
```

Comment:
Rename this to `collectAddressablePages` and move to:

```diff
  this.readSiteConfig(baseUrl)
+   .then(() => this.collectAddressablePages())
    .then(() => this.collectBaseUrl())
```

Let's preserve the "shape" of pages so we don't do this in `Site.prototype.generatePages`:

```js
const index = this.siteConfig.pages.findIndex(currentPage => currentPage.src === src);
```